azurelinuxagent/common/osutil/nsbsd.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright 2018 Stormshield
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import azurelinuxagent.common.utils.fileutil as fileutil
+import azurelinuxagent.common.utils.shellutil as shellutil
+import azurelinuxagent.common.utils.textutil as textutil
+import azurelinuxagent.common.logger as logger
+from azurelinuxagent.common.exception import OSUtilError
+from azurelinuxagent.common.osutil.freebsd import FreeBSDOSUtil
+from azurelinuxagent.common.future import ustr
+import azurelinuxagent.common.conf as conf
+import os
+import time
+
+class NSBSDOSUtil(FreeBSDOSUtil):
```

Comment:
It isn't clear to me if this code only supports your flavor of NSBSD or all flavors of NSBSD (if such a thing exists). Is Stormshield the only purveyor of this distro?

Reply:
Yes, NSBSD is not a public distro and Stormshield is the only purveyor.